data/topic-filter.go

```diff
@@ -0,0 +1,213 @@
+	EthnicGroupNationalIdentityAndReligion = Topic{
+		LocaliseKeyName: "EthnicGroupNationalIdentityAndReligion",
+		Group:           "ethnic_group_national_identity_and_religion",
+		SubTopics:       []string{"ethnic_group_national_identity_and_religion"},
+		ShowInWebUI:     true,
+	}
+
+	// HealthDisabilityAndUnpaidCare - Search information specific for health, disabilities and unpaid care
+	HealthDisabilityAndUnpaidCare = Topic{
+		LocaliseKeyName: "HealthDisabilityAndUnpaidCare",
+		Group:           "health_disability_and_unpaid_care",
+		SubTopics:       []string{"health_disability_and_unpaid_care"},
+		ShowInWebUI:     true,
+	}
+
+	// Housing - Search information specific for Housing
+	Housing = Topic{
+		LocaliseKeyName: "Housing",
+		Group:           "housing",
+		SubTopics:       []string{"housing"},
+		ShowInWebUI:     true,
+	}
+
+	// LabourMarketAndTravelToWork - Search information specific for labour market and travel to work
+	LabourMarketAndTravelToWork = Topic{
+		LocaliseKeyName: "LabourMarketAndTravelToWork",
+		Group:           "labour_market_and_travel_to_work",
+		SubTopics:       []string{"labour_market_and_travel_to_work"},
+		ShowInWebUI:     true,
+	}
+
+	// SexualOrientationAndGenderIdentity - Search information specific for sexual orientation and gender identity
+	SexualOrientationAndGenderIdentity = Topic{
+		LocaliseKeyName: "SexualOrientationAndGenderIdentity",
+		Group:           "sexual_orientation_and_gender_identity",
+		SubTopics:       []string{"sexual_orientation_and_gender_identity"},
+		ShowInWebUI:     true,
+	}
+
+	// Veterans - Search information specific for Veterans
+	Veterans = Topic{
+		LocaliseKeyName: "Veteran",
+		Group:           "veterans",
+		SubTopics:       []string{"veterans"},
+		ShowInWebUI:     true,
+	}
+
+	// topicFilterOptions contains all the possible filter available on the search page
+	topicFilterOptions = map[string]Topic{
+		DemographyAndMigration.Group:                 DemographyAndMigration,
+		Education.Group:                              Education,
+		EthnicGroupNationalIdentityAndReligion.Group: EthnicGroupNationalIdentityAndReligion,
+		HealthDisabilityAndUnpaidCare.Group:          HealthDisabilityAndUnpaidCare,
+		Housing.Group:                                Housing,
+		LabourMarketAndTravelToWork.Group:            LabourMarketAndTravelToWork,
+		SexualOrientationAndGenderIdentity.Group:     SexualOrientationAndGenderIdentity,
+		Veterans.Group:                               Veterans,
+	}
+)
+
+// reviewFilter retrieves filters from query, checks if they are one of the filter options, and updates validatedQueryParams
+func reviewTopicFilters(ctx context.Context, urlQuery url.Values, validatedQueryParams *SearchURLParams) error {
+	topicFilters := urlQuery["topics"]
+
+	if topicFilters == nil {
+		return nil
+	}
+
+	topics := strings.Split(topicFilters[0], ",")
+
+	for _, topicFilterQuery := range topics {
```

Comment:
Another for loop which could do with memory usage improvement, suggest doing the following:

```suggestion
	for i := range topics {
```

Where `topicFilterQuery` is being used in function to change this to `topics[i]`